planner/core/rule_partition_processor.go

```diff
@@ -313,27 +314,27 @@ func intersectionRange(start, end, newStart, newEnd int) (int, int) {
 }
 
 func (s *partitionProcessor) pruneRangePartition(ds *DataSource, pi *model.PartitionInfo) (LogicalPlan, error) {
-	var col *expression.Column
-	var fn *expression.ScalarFunction
-	var err error
-	// Partition by range.
-	if len(pi.Columns) == 0 {
-		col, fn, err = makePartitionByFnCol(ds.ctx, ds.TblCols, ds.names, pi.Expr)
-	} else if len(pi.Columns) == 1 {
-		// Convert partition by range columns to range.
-		col, fn, err = makePartitionByFnCol(ds.ctx, ds.TblCols, ds.names, pi.Columns[0].L)
+	// Partition by range columns.
+	if len(pi.Columns) > 0 {
+		return s.pruneRangeColumnsPartition(ds, pi)
```

Comment:
should we let "range column by int" like this can use `makeLessThanData`?
```
create table t (a varchar(32)) partition by range columns (a) (
    partition p0 values less than (0),
    partition p1 values less than (500),
    partition p2 values less than (1000),
    partition p3 values less than (1500),
    partition p4 values less than (2000)
);
select * from t;
```

(that way should be faster?)

Reply:
Yes, it's faster, but the case is not general, we have to write more code for this optimizing